### PR TITLE
convert histogram metrics

### PR DESCRIPTION
**Pull request by Keksoj**
1. Sōzu now aggregates metrics accross workers, no need to do it anymore
2. Sōzu now produces prometheus-compatible histograms, we can convert them to metric lines.

### CODE REVIEW

File: src/svc/telemetry/prometheus.rs [hunk 1]
```diff
@@ -50,33 +50,85 @@ impl LabeledMetric {
...
 
+    /// Format labels in a comma-separated list:
+    ///
+    /// ```plain
+    /// "label"="value","other"="value"
+    /// ```
+    fn formatted_labels(&self) -> String {
+        self.labels
+            .iter()
+            .map(|(name, value)| format!("{}=\"{}\"", name, value))
```
Comment (llenotre) [inline src/svc/telemetry/prometheus.rs:69]: Use `Iterator`'s `chain` instead of `format!`

File: src/svc/telemetry/prometheus.rs [hunk 2]
```diff
@@ -50,33 +50,85 @@ impl LabeledMetric {
...
+                        lines.push_str(&format!(
+                            "{}_sum{{{}}} {}\n",
+                            printable_metric_name, formatted_labels, hist.sum
+                        ));
+                        lines.push_str(&format!(
+                            "{}_count{{{}}} {}",
+                            printable_metric_name, formatted_labels, hist.count
+                        ));
+
+                        lines
```
Comment (llenotre) [inline src/svc/telemetry/prometheus.rs:122]: Iterator, chain, collect

File: src/svc/telemetry/prometheus.rs [hunk 3]
```diff
@@ -125,10 +175,7 @@ pub fn convert_metrics_to_prometheus(
 }
 
 /// assign worker_id and cluster_id as labels
-fn apply_labels(
-    aggregated_metrics: AggregatedMetrics,
-    aggregate_backend_metrics: bool,
-) -> Vec<LabeledMetric> {
+fn apply_labels(aggregated_metrics: AggregatedMetrics) -> Vec<LabeledMetric> {
     let mut labeled_metrics = Vec::new();
```
Comment (llenotre) [inline src/svc/telemetry/prometheus.rs:179]: Iterator, flat_map, collect

File: src/svc/telemetry/prometheus.rs (hunk 2 above)
Comment (Keksoj) [reply src/svc/telemetry/prometheus.rs]: chain is used ;-)

File: src/svc/telemetry/prometheus.rs (hunk 1 above)
Comment (Keksoj) [reply src/svc/telemetry/prometheus.rs:69]: for two items it is not yet worth it

File: src/svc/telemetry/prometheus.rs (hunk 3 above)
Comment (Keksoj) [reply src/svc/telemetry/prometheus.rs:177]: we will optimize the code if we find this to be a bottleneck in production

**Review by llenotre — APPROVED** [review]
lgtm

**Pull request merged**